File: java/404.sum-of-left-leaves.java
/*
 * @lc app=leetcode id=404 lang=java
 *
 * [404] Sum of Left Leaves
 */
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        if (root == null) return 0;
        TreeNode leftNode = root.left;
        int leftSum = leftNode != null && 
            leftNode.left == null && leftNode.right == null ? 
            leftNode.val : sumOfLeftLeaves(root.left);
        int rightSum = sumOfLeftLeaves(root.right);
        return leftSum + rightSum;
    }
}

File: go/404.sum-of-left-leaves.go
/*
 * @lc app=leetcode id=404 lang=golang
 *
 * [404] Sum of Left Leaves
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumOfLeftLeaves(root *TreeNode) int {
	return recal(root, false)
}

func recal(root *TreeNode, isLeft bool) int {
	if root == nil {
		return 0
	}
	if isLeft && root.Left == nil && root.Right == nil {
		return root.Val
	}
	return recal(root.Left, true) + recal(root.Right, false)
}
// @lc code=end

